bluesky/callbacks/fitting.py

```diff
@@ -290,5 +294,35 @@ def compute(self):
             if len(_cen_list) >= 2:
                 self.fwhm = np.abs(self.crossings[-1] - self.crossings[0],
                                    dtype=float)
+
+        # Calculate the derivative of the data
+        x_der = x[1:]
+        y_der = np.diff(y)
+        self.der.x = x_der
+        self.der.y = y_der
+        max_der = np.argmax(y_der)
+        min_der = np.argmin(y_der)
+        self.der.max = x[max_der], self.y_data[max_der],
+        self.der.min = x[min_der], self.y_data[min_der],
+        self.der.com, = np.interp(center_of_mass(y), np.arange(len(x_der)), x_der)
```

Comment:
Don't you want `center_of_mass(y_der)` here?

Reply:
You are right.

I tried to incorporate a der flag in compute with a default value being false, so it will compute the derivative only if specify to be True. What do you think?

How can I test this at the beamline (not a big expert with git). Should I create a specific environement at the beamline and install git+my forked bluesky?

Reply:
Answered via Slack about testing.